### ISSUE TITLE
prometheus the ALERTING RULES whether to provide the API?

### ISSUE THREAD

**Issue by xiaoqge** [prometheus/prometheus#1508]
prometheus the ALERTING RULES whether to provide the API? We want to do some renovation to the user profile page in WEB UI in alerting rules. Please help me, thank you

**Comment by beorn7** [comment]
Sorry, I don't really understand what you want to tell us. Could you rephrase?

**Comment by xiaoqge** [comment]
@beorn7 I'm sorry, I have described is not clear enough, I want to use api to modify the form of alerting rules, rather than by modifying the rule file, there are ways?

**Comment by beorn7** [comment]
We don't do configuration via API in the Prometheus ecosystem. Configuration must be in files. If you want to change configuration via an API, you have to use a config management system that generates config files based on API calls.

This is the same argument as @brian-brazil gave you in https://github.com/prometheus/prometheus/issues/1505 .

**Comment by grobie** [comment]
This might be of interest https://github.com/kelseyhightower/confd
